### PR TITLE
Synchronization - Clean subpass dependency init

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Fix incorrect subpass dependency initialization -- addressing validation issue.

### CODE REVIEW

File: drivers/vulkan/rendering_device_vulkan.cpp [hunk 1]
```diff
@@ -3227,8 +3227,8 @@ VkRenderPass RenderingDeviceVulkan::_render_pass_create(const Vector<AttachmentF
 		}
 		if (reference.layout != description.finalLayout) {
 			// NOTE: this should be smarter based on the textures knowledge of it's subsequent role
-			dependency_from_external.dstStageMask |= VK_PIPELINE_STAGE_ALL_COMMANDS_BIT;
-			dependency_from_external.dstAccessMask |= VK_ACCESS_MEMORY_READ_BIT | VK_ACCESS_MEMORY_WRITE_BIT;
+			dependency_to_external.dstStageMask |= VK_PIPELINE_STAGE_ALL_COMMANDS_BIT;
```
Comment (jzulauf-lunarg) [inline drivers/vulkan/rendering_device_vulkan.cpp:3230]: Cut and paste error.  Ugh.

**Comment by jzulauf-lunarg** [comment]
@reduz  -- I noticed one of my changes was disabled... then I noticed it was my fault...

**Review by akien-mga — APPROVED** [review]
Approved by @reduz on IRC.

**Comment by akien-mga** [comment]
Thanks!

**Comment by jzulauf-lunarg** [comment]
@akien-mga @reduz -- the honors are still about even.  Fixing that bug exposed a critical issue in my next major update... so godot continues to be a great test case for me.

**Pull request merged**